llvm/lib/Target/AArch64/AArch64AsmPrinter.cpp

```diff
@@ -2277,28 +2277,39 @@ void AArch64AsmPrinter::LowerLOADgotAUTH(const MachineInstr &MI) {
   const MachineOperand &GAMO = MI.getOperand(1);
   assert(GAMO.getOffset() == 0);
 
-  MachineOperand GAHiOp(GAMO);
-  MachineOperand GALoOp(GAMO);
-  GAHiOp.addTargetFlag(AArch64II::MO_PAGE);
-  GALoOp.addTargetFlag(AArch64II::MO_PAGEOFF | AArch64II::MO_NC);
+  if (MI.getMF()->getTarget().getCodeModel() == CodeModel::Tiny) {
```

Comment:
@ilovepi Thanks for your suggestion from https://github.com/llvm/llvm-project/pull/113812#discussion_r1823353946! Applied here.